### ISSUE TITLE
[exchange/365] cannot see the global contacts of the organization 

### ISSUE THREAD

**Issue by ftoledo** [jobisoft/TbSync#620]
## Your environment

TbSync version: 4.3
Thunderbird version: 102.4.0 (Linux)

## Expected behavior
Global adddressbook from 365 account must be show and search on new mail 
...

## Actual behavior
...
it is showing only personal contacts
![imagen](https://user-images.githubusercontent.com/95441/197226210-e938486d-afcf-465a-a247-82b7f803fecb.png)

![imagen](https://user-images.githubusercontent.com/95441/197226451-feb69ba2-2792-46bb-87d0-bae7ba03529b.png)

since updating tb to 102 the global contacts of 365 of my organization do not appear. In previous versions I could search the entire directory

![imagen](https://user-images.githubusercontent.com/95441/197227182-ed925c95-9a66-4142-a8ee-6f934a1773cb.png)

**Comment by jobisoft** [comment]
No longer implemented. Will come back eventually.

**Comment by ftoledo** [comment]
hi @jobisoft is this related to the fact that thunderbird carddav still does not support contact groups?

**Comment by jobisoft** [comment]
No. I threw it out to speed up the update process from 91 to 102. I simply need more time to get that feature back in.

**Comment by ftoledo** [comment]
Excellent! I'll wait for the feature to come back. Thank you for your work

**Comment by jmechnich** [comment]
Is there a rough timeline for this?

**Comment by ragdollfun** [comment]
Is there a way we can help ?

**Comment by ftoledo** [comment]
any news for this issue? i miss the global address searches =(

**Comment by garv-shah** [comment]
i'd also love for this feature to be added :)

**Comment by TomJansen** [comment]
I really wish for this to go back, it is the only thing that is needed for me to drop using Owl

**Comment by Ch1llb0** [comment]
> I know this is still pending, but I decided to concentrate on my other add-on Quicktext for now. I have made huge progress converting it to a pure WebExtension ([jobisoft/quicktext#371](https://github.com/jobisoft/quicktext/issues/371)). It will be my top priority to finish that.
> 
> Once done, I will convert TbSync and only after that has finished, I will be able to bring back the global address book lookup.

I believe, the conversion of your other add-on has been finished... Any chance we can see work towards bringing back GAL? :)

**Comment by qwertychouskie** [comment]
By any chance do you happen to have an update of when you think this may be fixed?  I was confused why the option I had seen in screenshots online wasn't showing, then I came across this issue.

**Comment by DevDef** [comment]
Given that the Mozilla Team dedicated to Thunderbird is announcing to support Exchange protocols in the coming months, I doubt that John will ever work on the missing/broken features in his extension.
https://blog.thunderbird.net/2024/04/adventures-in-rust-bringing-exchange-support-to-thunderbird/
The bad news is that the global address book will come in a second time:
> _Until recently, this functionality was only available through an add-on. But, in the next ESR (Extended Support) release of Thunderbird in July 2024, we expect to provide this support natively within Thunderbird. Because of the size of this undertaking, the first roll-out of the Exchange support will [initially cover only email](https://youtu.be/7jNV1J2pdPc), with calendar and address book support coming at a later date._

**Comment by je-vv** [comment]
The thing with TB, is that it's not clear how they'll integrate with o365, besides having delayed such integration.  I would expect TB to expose the calendar as caldav, and the address book as cardav, just like TbSync does, only more integrated, but I have no clue.  Also regarding messages, I wouldn't expect any change, I still want it to be IMAP, as I get my emails and send them right now, so regarding messages I don't see what is it they're doing, but they announced changes there as well.  So we need to see what's that integration, and what sort of things it depends on, and if it adds weird not common or open source solutions (in the end it's microsoft stuff the one being integrated).

All that to say, perhaps TbSync will remain a need depending of what TB integration means, which is yet to be seen, at least for those not having the time to explore nightly builds and testing/unstable stuff...

**Comment by jobisoft** [comment]
> The thing with TB, is that it's not clear how they'll integrate with o365

They are adding support for EWS. The bigger counterpart of EAS (which TbSync uses).

> besides having delayed such integration.

Who delayed what? They announced EWS support in the second half of last year, and they are probably going to have it shipped with ESR128 in a few weeks. That is an extremely short integration cycle.

> I would expect TB to expose the calendar as caldav, and the address book as cardav, just like TbSync does, only more integrated, but I have no clue.

Something along those lines, yes. But it will not be part of the initial implementation. But it is planned.

> Also regarding messages, I wouldn't expect any change, I still want it to be IMAP, as I get my emails and send them right now, so regarding messages I don't see what is it they're doing

They are adding native EWS support. If your tenant happens to disable IMAP (which many do), you are stuck. You currently need an add-on (owl or exquila) to get your emails.

> if it adds weird not common or open source solutions (in the end it's microsoft stuff the one being integrated).

Yes. That is why Thunderbird did not want to integrate it for so long, because it is not a standard. But the council agreed that freedom is also "Freedom of Choice": If your tenant is disabling IMAP, you should not be forced to use Outlook. You should still have the choice to use a different mail client.
 
> All that to say, perhaps TbSync will remain a need depending of what TB integration means, which is yet to be seen, at least for those not having the time to explore nightly builds and testing/unstable stuff...

I will deprecate TbSync support for O365 soon. It was intended for EAS only, aimed at supporting 3rd party servers, like Zimbra. But since O365 also supported it, many did use it for that (and I did add OAuth support just for them). Once calendar and address book support for EWS is natively added, I will pull the plug.

**Comment by Giermann** [comment]
> Once calendar and address book support for EWS is natively added, I will pull the plug.

Let us all hope this will happen a bit before October 2026, when MS drops EWS support in Exchange Online.
And/or that Thunderbird will add support for protocols that are being kept in O365.

I personally currently use TB/TbSync with an IMAP enabled O365 tenant, although it's not working perfect.
And it really shivers me to hear, that I'll be forced to use Outlook in less than 30 months...

**Comment by qwertychouskie** [comment]
Does anyone know what the status of adding address book support natively to Thunderbird is?  It's been almost exactly 1 year since the last communication/comments here, and it doesn't look like any of the newer TB blogs address this either.

**Comment by jobisoft** [comment]
The feature is available again: https://github.com/jobisoft/EAS-4-TbSync/issues/309
